### ISSUE TITLE
Are action functions isolated?

### ISSUE THREAD

**Issue by sharmapukar217** [rakkasjs/rakkasjs#56]
When i call my api from action functions and try to access or set cookies, no cookies are sent to server and neither the server sets the cookie.  But it works well and cookies are set if I use fetch. [Here](https://github.com/sharmapukar217/rakkas-js-no-credentials) is the repo set up to reproduce the issue

**Comment by cyco130** [comment]
Hi,

I think the confusion comes from the fact that `action` **always runs on the server**, therefore it is isolated from the client. You don't have to call API routes from it, you can just access your database etc. from within the `action` function. And you can set cookies from it using `ctx.requestContext.setCookie("aaa", "bbb")`.

`ctx.fetch` is just like `node-fetch` except:

1. You can use relative URLs (so the `ctx.url.origin +` bit isn't necessary).
2. Requests to your own routes are made directly without the network roundtrip.
3. Credentials are forwarded (unless you omit them with `credentials: "omit"`).

But the cookies in the response are not automatically passed on to the response, you can control the response headers yourself if you want, though:

```js
export async function action(ctx) {
  const response = await ctx.fetch("/test?from=useSubmit", {
    method: "POST",
    credentials: "include",
  });

  return {
    data: {},
    headers: {
      "Set-Cookie": response.headers.getSetCookie(),
    },
  };
}
```

`getSetCookie` is a [new API proposal](https://github.com/whatwg/fetch/pull/1346) for getting multiple `Set-Cookie` headers from a `Headers` object. HatTip polyfills it.

Also, you don't have to inject the same middleware in `beforePages` and `beforeApiRoutes` separately (doing so sets two cookies actually!). Rakkas looks for pages, API routes, and not found handlers in that order. Middlewares injected in one step will be available in the next steps.

**Comment by sharmapukar217** [comment]
and what about sending cookie to server? 😕

**Comment by cyco130** [comment]
A POST request is sent to the server when you submit and it will contain currently set cookies. You can access them with `ctx.requestContext.cookie`.

What's _not_ working is that it should be forwarding those credentials when you use `ctx.fetch` but it isn't. It works on Node 14 with `node-fetch` but Node16+ has native fetch that ignores the `include` flag. Will fix it, thanks for the report.

**Comment by cyco130** [comment]
> What's _not_ working is that it should be forwarding those credentials when you use `ctx.fetch` but it isn't. It works on Node 14 with `node-fetch` but Node16+ has native fetch that ignores the `include` flag. Will fix it, thanks for the report.

Fixed in `0.6.2-next.5` (`npm install -D rakkasjs@next`).

**Comment by sharmapukar217** [comment]
> > What's _not_ working is that it should be forwarding those credentials when you use `ctx.fetch` but it isn't. It works on Node 14 with `node-fetch` but Node16+ has native fetch that ignores the `include` flag. Will fix it, thanks for the report.
> 
> Fixed in `0.6.2-next.5` (`npm install -D rakkasjs@next`).
TYSM for this fix ❤️
